### PR TITLE
feat: allow setting args for variadic functions

### PR DESCRIPTION
**Pull request by bergotorino**
The gomock's Call interface allows for setting a value to a specified
argument, making it convinient to work with functions that return data
through pointer arguments. However, this is not possible for variadic
functions, the error is:

  SetArg(1, ...) called for a method with 1 args...

The root cause is that the variadic argument in a form of
...interface{} is threated as single value instead of being treated as
slice.

This commit changes this situation by making sure that the variadic
arguments that are a slice are treated as slices. As a consequence it is
now possible to set argument value for variadic functions.

See: https://github.com/golang/mock/issues/174

### CODE REVIEW

File: gomock/call.go [hunk 1]
```diff
@@ -225,15 +225,17 @@ func (c *Call) SetArg(n int, value interface{}) *Call {
 	c.t.Helper()
 
 	mt := c.methodType
-	// TODO: This will break on variadic methods.
-	// We will need to check those at invocation time.
-	if n < 0 || n >= mt.NumIn() {
+	if (n < 0 || n >= mt.NumIn()) && !mt.IsVariadic() {
```
Comment (codyoss) [inline gomock/call.go:228]: Could you please add some tests for this. You can either do a traditional tests or a sample code generation like others in the `mockgen/internal/test` dir.

**Review by codyoss — CHANGES_REQUESTED** [review]
Please add tests :)

**Comment by bergotorino** [comment]
Sure, will do. Thanks for review and sorry for missing your comments on time

**Comment by mvrahden** [comment]
@bergotorino are you still planning to do this?

**Comment by codyoss** [comment]
Closing due to lack of response.

**Pull request closed without merging**